docs/source/tutorial/peft_model_config.md

```diff
@@ -0,0 +1,169 @@
+<hfoption id="p-tuning">
+
+```json
+{
+  "base_model_name_or_path": "roberta-large", #base model to apply p-tuning to
+  "encoder_dropout": 0.0,
+  "encoder_hidden_size": 128,
+  "encoder_num_layers": 2,
+  "encoder_reparameterization_type": "MLP",
+  "inference_mode": true,
+  "num_attention_heads": 16,
+  "num_layers": 24,
+  "num_transformer_submodules": 1,
+  "num_virtual_tokens": 20,
+  "peft_type": "P_TUNING", #PEFT method type
+  "task_type": "SEQ_CLS", #type of task to train model on
+  "token_dim": 1024
+}
+```
+
+You can create your own configuration for training by initializing a [`PromptEncoderConfig`].
+
+```py
+from peft import PromptEncoderConfig
+
+p_tuning_config = PromptEncoderConfig(
+    encoder_reprameterization_type="MLP",
+    encoder_hidden_size=128,
+    num_attention_heads=16,
+    num_layers=24,
+    num_transformer_submodules=1,
+    num_virtual_tokens=20,
+    token_dim=1024,
+    task_type="SEQ_CLS"
+)
+```
+
+</hfoption>
+</hfoptions>
+
+## PEFT models
+
+With a configuration for a PEFT method in hand, you can now apply it to any pretrained model to create a [`PeftModel`]. You can choose from any of the state-of-the-art models from the Transformers library, a custom model, and even new and unsupported transformer architectures.
+
+Load a base model to finetune.
+
+```py
+from transformers import AutoModelForCausalLM
+
+model = AutoModelForCausalLM.from_pretrained(
+    "facebook/opt-350m"
+)
+```
+
+Create a [`PeftModel`] from the base model and a configuration (for this tutorial, you'll use the [`LoraConfig`] you created earlier). The [`get_peft_model`] method returns a trainable [`PeftModel`].
+
+```py
+from peft import get_peft_model
+
+lora_model = get_peft_model(model, lora_config)
+lora_model.print_trainable_parameters()
+"trainable params: 1,572,864 || all params: 332,769,280 || trainable%: 0.472659014678278"
+```
+
+Now you can train the [`PeftModel`] with your preferred training framework!
+
+After you've trained a [`PeftModel`], you can save it locally or upload it to the Hub. To load a [`PeftModel`], you'll need to provide the [`PeftConfig`] used to create it and the base model it was trained from.
+
+```py
+from peft import PeftModel, PeftConfig
+
+config = PeftConfig.from_pretrained("ybelkada/opt-350m-lora")
+model = AutoModelForCausalLM.from_pretrained(config.base_model_name_or_path)
+lora_model = PeftModel.from_pretrained(model, "ybelkada/opt-350m-lora")
+```
+
+By default, the [`PeftModel`] can only be used for inference, but if you'd like to train the adapter some more you can set `is_trainable=True`.
+
+```py
+lora_model = PeftModel.from_pretrained(model, "ybelkada/opt-350m-lora", is_trainable=True)
+```
+
+The [`PeftModel.from_pretrained`] method is the most flexible way to load a [`PeftModel`] because it doesn't matter what model framework was used (Transformers, timm, a generic PyTorch model). Other classes, like [`AutoPeftModel`], are just a convenient wrapper around the base [`PeftModel`].
```

Comment:
```suggestion
The [`PeftModel.from_pretrained`] method is the most flexible way to load a [`PeftModel`] because it doesn't matter what model framework was used (Transformers, timm, a generic PyTorch model). Other classes, like [`AutoPeftModel`], are just a convenient wrapper around the base [`PeftModel`], and makes it easier to load PEFT models directly using the hub id or the local path where you stored your PEFT weights:

```py
from peft import AutoPeftModelForCausalLM

lora_model = AutoPeftModelForCausalLM.from_pretrained("ybelkada/opt-350m-lora")
```
Read more about it in the [auto classes section](../packages_reference/auto_class).
```